frontend/components/data-table.js

```diff
@@ -0,0 +1,550 @@
+      this._clearCheckButtonsStyle();
+
+      const btn = event.target.closest('[data-type="filter-selected"]');
+      btn.classList.add('header-button-active');
+      const selectedFilter = btn.dataset.value;
+
+      this._updateState({ selectedFilter });
+      this._resetCurrentPage();
+      this._makeFiltering();
+
+      const { itemsPerPage, buttonsCount } = this._state;
+      const { phonesLength } = this._props;
+
+      this._pagination.updateProps({
+        perPage: itemsPerPage,
+        phonesLength,
+        buttonsCount,
+      });
+    });
+
+    this.on('input', '[data-element="find-input"]', () => {
+      this.emit('input-enter');
+    });
+
+    this.on('click', '[data-sortable-key]', (event) => {
+      const element = event.target.closest('[data-sortable-key]');
+      const currOrderName = element.dataset.sortableKey;
+
+      let { orderName, orderValue } = this._state;
+
+      if (orderName === currOrderName) {
+        orderValue = orderValue === 'up' ? 'down' : 'up';
+      } else {
+        orderName = currOrderName;
+      }
+
+      this._updateState({ orderValue, orderName });
+      this.emit('order-enter');
+    });
+
+    this.subscribe(
+      'input-enter',
+      this._debounce(() => {
+        this._resetCurrentPage();
+        this._makeFiltering();
+        const {
+          currentPage,
+          itemsPerPage,
+          buttonsCount,
+        } = this._state;
+
+        const { phonesLength } = this._props;
+        this._pagination.updateProps({
+          currentPage,
+          itemsPerPage,
+          phonesLength,
+          buttonsCount,
+        });
+      }, 500),
+    );
+
+    this.subscribe('order-enter', this._makeFiltering.bind(this));
+  }
+
+  _initPagination() {
+    const {
+      itemsPerPage,
+      buttonsCount,
+      currentPage,
+      perPageOptions
+    } = this._state;
+
+    const { phonesLength } = this._props;
+
+    this._paginationSelector = new PaginationSelector({
+      element: this._getComponent('pagination-selector'),
+      props: {
+        perPageOptions,
+        itemsPerPage,
+      },
+    });
+
+    this._pagination = new Pagination({
+      element: this._getComponent('pagination'),
+      props: {
+        phonesLength,
+        itemsPerPage,
+        buttonsCount,
+        currentPage,
+      },
+    });
+
+    this._paginationSelector.subscribe('change-per-page', (newPerPage) => {
+      this._updateState({ itemsPerPage: newPerPage });
+      this._makeFiltering();
+      const { buttonsCount } = this._state;
+
+      this._pagination.updateProps({
+        itemsPerPage: newPerPage,
+        buttonsCount,
+      });
+
+      this._paginationSelector.updateProps({
+        itemsPerPage: newPerPage,
+      });
+    });
+
+    this._pagination.subscribe('page-changed', (newPage) => {
+      this._updateState({ currentPage: newPage });
+      this._pagination.updateProps({ currentPage: newPage });
+
+      this._makeFiltering();
```

Comment:
don't do side effects
just update the state 
all the rest should be done in updateView